query/src/test/java/org/infinispan/query/affinity/RpcCollector.java

```diff
@@ -0,0 +1,38 @@
+package org.infinispan.query.affinity;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.infinispan.statetransfer.StateRequestCommand;
+import org.infinispan.statetransfer.StateResponseCommand;
+
+class RpcCollector {
+
+   private List<RpcDetail> rpcCollection = new ArrayList<>();
+
+   private Predicate<RpcDetail> stateTransferPredicate = rpc ->
+         Arrays.asList(StateRequestCommand.class, StateResponseCommand.class).contains(rpc.getCommand().getClass());
```

Comment:
functionalicious? :smile:

It will allocate a list every time the predicate is evaluated, let me sort it out